HC/WrapperGen/WrapperGen.cpp

```diff
@@ -284,21 +289,28 @@ struct StringFinder
       switch(rop) {
         // type function(type1 arg1, type1 arg2, type2 arg3)
         case PARAMETERS:
+          if(i->getTypeName() == "grid_launch_parm_cxx") {
+              //out << "const ";
+          }
           out << i->getTypeNameWithQual() << " " << i->getArgName();
           break;
         // type function(grid_launch_parm lp, type1 arg2, type2 arg3)
         // retain grid_launch_parm for compatibility
         case PARAMETERSKERNEL:
           if(i->getTypeName() == "grid_launch_parm_cxx")
-            out << "grid_launch_parm" << " " << i->getArgName();
+            out << "const grid_launch_parm" << " " << i->getArgName();
           else
             out << i->getTypeNameWithQual() << " " << i->getArgName();
           break;
         // Class::Class(grid_launch_parm _lp, type1 arg2, type2 arg3)
         case PARAMETERSCONSTR:
-          out << i->getTypeNameWithQual() << " ";
-          if(i->getTypeName() == "grid_launch_parm_cxx")
-            out << "_";
+          if(i->getTypeName() == "grid_launch_parm_cxx") {
+            out << "const " ;
+            out << i->getTypeNameWithQual() << " ";
+            out << "x";
```

Comment:
Yes exactly.

On Jun 22, 2016, at 8:02 PM, Wen-Heng (Jack) Chung <notifications@github.com<mailto:notifications@github.com>> wrote:


In HC/WrapperGen/WrapperGen.cpp<https://github.com/RadeonOpenCompute/hcc/pull/80#discussion_r68161740>:

>            else
>              out << i->getTypeNameWithQual() << " " << i->getArgName();
>            break;
>          // Class::Class(grid_launch_parm _lp, type1 arg2, type2 arg3)
>          case PARAMETERSCONSTR:
> -          out << i->getTypeNameWithQual() << " ";
> -          if(i->getTypeName() == "grid_launch_parm_cxx")
> -            out << "_";
> +          if(i->getTypeName() == "grid_launch_parm_cxx") {
> +            out << "const " ;
> +            out << i->getTypeNameWithQual() << " ";
> +            out << "x";


this is to differentiate "_lp" from "__lp", right? now it becomes "_lp" and "x_lp".

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/RadeonOpenCompute/hcc/pull/80/files/758b7f57d57a3a927dd454b7d8011318ecb129f0#r68161740>, or mute the thread<https://github.com/notifications/unsubscribe/ACYSAv4Twq6E3uCFguNeXxcMmG5d4nBPks5qOdsGgaJpZM4I8LAJ>.